feat: enhance logging in notification handling

joel-medicala-yral opened the pull request:
- Added detailed logging for the send_notification function to track user actions and authentication status.
- Improved logging in send_message_to_group to capture notification key and data payload details.
- This enhancement aids in debugging and monitoring notification processes.

closes https://github.com/dolr-ai/product-roadmap/issues/332

uwuxyz committed:
feat: enhance logging in notification handling

- Added detailed logging for the send_notification function to track user actions and authentication status.
- Improved logging in send_message_to_group to capture notification key and data payload details.
- This enhancement aids in debugging and monitoring notification processes.

uwuxyz committed:
refactor: clarify notification key variable naming

- Renamed `notification_key` to `notification_key_from_firebase` for better clarity in the context of device registration.
- Updated the logic in `update_notification_devices` to check for the specific JSON structure in the remove operation.

uwuxyz committed:
refactor: simplify remove operation check in update_notification_devices

- Updated the logic in `update_notification_devices` to check for the presence of the string "remove" instead of the full JSON structure. This change streamlines the operation detection process.

uwuxyz committed:
feat: enhance device registration logic with Firebase integration

- Implemented a new method `get_notification_key_by_name` in the Firebase module to retrieve notification keys by name.
- Updated the `register_device` function to handle both adding and creating notification keys based on the presence of existing keys in user metadata.
- Improved error handling and logging throughout the registration process for better traceability and debugging.

uwuxyz committed:
refactor: streamline device registration process and improve error handling

- Simplified the `register_device` function by removing unnecessary unwraps and enhancing error handling for metadata retrieval.
- Consolidated logic for updating notification keys and registration tokens, ensuring clearer flow and better logging.
- Removed the unused `get_notification_key_by_name` method from the Firebase module to reduce code clutter.

uwuxyz committed:
feat: enhance device registration and Firebase integration

- Updated the `register_device` function to improve handling of notification keys, including logic for adding, removing, and creating keys based on user metadata and Firebase state.
- Introduced a new method `get_notification_key` in the Firebase module to retrieve existing notification keys, enhancing the registration process.
- Improved error handling and logging throughout the device registration flow for better traceability and debugging.

uwuxyz committed:
refactor: optimize device registration logic and Firebase interaction

- Streamlined the `register_device` function by removing unnecessary clones and improving the handling of registration tokens.
- Enhanced error handling for metadata retrieval, returning appropriate errors when metadata is not found.
- Updated the `update_notification_devices` method to simplify response handling and improve logging for better traceability.

uwuxyz committed:
refactor: enhance device registration logic and Firebase interaction

- Renamed variables for clarity and improved handling of registration tokens to avoid conflicts.
- Streamlined the logic for adding, removing, and creating notification keys based on user metadata and Firebase state.
- Enhanced error handling and logging throughout the `register_device` function for better traceability and debugging.

uwuxyz committed:
refactor: improve device registration logic and Firebase interaction

- Renamed variables for clarity and streamlined the handling of registration tokens in the `register_device` function.
- Enhanced error handling for metadata retrieval and improved the logic for adding and creating notification keys based on user metadata.
- Updated the response handling from Firebase to ensure proper error propagation and logging for better traceability.

komal-sai-yral commented:
@joel-medicala-yral feel free to integrate sentry if required

ravi-sawlani-yral commented on server/src/notifications.rs:
```diff
@@ -201,32 +204,44 @@ async fn send_notification(
     let user_metadata: UserMetadata = serde_json::from_slice(&meta_raw).map_err(Error::Deser)?;
 
     let Some(notification_key) = user_metadata.notification_key else {
+        log::warn!("[send_notification] Notification key not found for user: {}", user);
         return Ok(Json(Err(ApiError::NotificationKeyNotFound)));
     };
+    log::info!("[send_notification] Notification key found for user: {}: {}", user, notification_key.key);
 
     let data = req.0.data;
+    log::info!("[send_notification] Preparing to send data for user {}: {:?}", user, data);
 
+    log::info!("[send_notification] Calling send_message_to_group for user: {}", user);
     state
         .firebase
         .send_message_to_group(notification_key, data)
         .await?;
+    log::info!("[send_notification] Successfully sent/processed notification for user: {}", user);
```
Should we club these. Looks like debug logs

joel-medicala-yral committed:
refactor: streamline device registration token handling (#21)

- Updated the `register_device` and `unregister_device` functions to use only the registration token for identifying devices, removing the reliance on device fingerprints.
- Simplified the logic for adding and removing registration tokens, enhancing clarity and maintainability.
- Adjusted the `DeviceRegistrationToken` struct to remove the device fingerprint field, reflecting the updated registration logic.

Co-authored-by: JoeruCodes <joymedi8143@gmail.com>

joel-medicala-yral committed:
Joel/fingerprint removal (#22)

* refactor: streamline device registration token handling

- Updated the `register_device` and `unregister_device` functions to use only the registration token for identifying devices, removing the reliance on device fingerprints.
- Simplified the logic for adding and removing registration tokens, enhancing clarity and maintainability.
- Adjusted the `DeviceRegistrationToken` struct to remove the device fingerprint field, reflecting the updated registration logic.

* chore: update dependencies and enhance device registration logic

- Upgraded several dependencies including `serde`, `candid`, and `futures` to their latest versions for improved performance and security.
- Introduced `yral-types` and `ic-agent` as workspace dependencies to streamline project structure.
- Refactored device registration functions to utilize `DelegatedIdentityWire`, enhancing the handling of identity verification and improving error handling.
- Added new `serdect` package for additional functionality related to serialization and deserialization.

---------

Co-authored-by: JoeruCodes <joymedi8143@gmail.com>

joel-medicala-yral committed:
Joel/fingerprint removal (#24)

* refactor: streamline device registration token handling

- Updated the `register_device` and `unregister_device` functions to use only the registration token for identifying devices, removing the reliance on device fingerprints.
- Simplified the logic for adding and removing registration tokens, enhancing clarity and maintainability.
- Adjusted the `DeviceRegistrationToken` struct to remove the device fingerprint field, reflecting the updated registration logic.

* chore: update dependencies and enhance device registration logic

- Upgraded several dependencies including `serde`, `candid`, and `futures` to their latest versions for improved performance and security.
- Introduced `yral-types` and `ic-agent` as workspace dependencies to streamline project structure.
- Refactored device registration functions to utilize `DelegatedIdentityWire`, enhancing the handling of identity verification and improving error handling.
- Added new `serdect` package for additional functionality related to serialization and deserialization.

* Revert "chore: update dependencies and enhance device registration logic"

This reverts commit 8196c3b2674d53ff7feb2faf51004373daf731f3.

---------

Co-authored-by: JoeruCodes <joymedi8143@gmail.com>

joel-medicala-yral committed:
Joel/fingerprint removal (#25)

* refactor: streamline device registration token handling

- Updated the `register_device` and `unregister_device` functions to use only the registration token for identifying devices, removing the reliance on device fingerprints.
- Simplified the logic for adding and removing registration tokens, enhancing clarity and maintainability.
- Adjusted the `DeviceRegistrationToken` struct to remove the device fingerprint field, reflecting the updated registration logic.

* chore: update dependencies and enhance device registration logic

- Upgraded several dependencies including `serde`, `candid`, and `futures` to their latest versions for improved performance and security.
- Introduced `yral-types` and `ic-agent` as workspace dependencies to streamline project structure.
- Refactored device registration functions to utilize `DelegatedIdentityWire`, enhancing the handling of identity verification and improving error handling.
- Added new `serdect` package for additional functionality related to serialization and deserialization.

* Revert "chore: update dependencies and enhance device registration logic"

This reverts commit 8196c3b2674d53ff7feb2faf51004373daf731f3.

---------

Co-authored-by: JoeruCodes <joymedi8143@gmail.com>

uwuxyz committed:
refactor: update device registration logic and clean up dependencies

- Refactored the `register_device` and `unregister_device` functions to utilize a signature for identity verification, removing reliance on `DelegatedIdentityWire`.
- Cleaned up the `Cargo.toml` and `Cargo.lock` files by removing unused dependencies such as `yral-types` and `serdect`.
- Updated the `serde` dependency version for consistency across the project.
- Enhanced error handling in the device registration process for improved clarity and maintainability.
